openedx/core/djangoapps/catalog/utils.py

```diff
@@ -76,6 +76,30 @@ def get_programs(user=None, uuid=None, type=None):  # pylint: disable=redefined-
         return []
 
 
+def get_programs_for_credentials(user, programs_credentials):
```

Comment:
I would move this function back to the `credentials` app. The `catalog` app shouldn't need to know about credentials, and processing Credentials API responses.